backends/smt2/smtbmc.py

```diff
@@ -1158,6 +1158,8 @@ def smt_forall_assert():
     global asserts_cache_dirty
     asserts_cache_dirty = False
 
+    assert (len(smt.modinfo[topmod].maximize) <= 1 and len(smt.modinfo[topmod].minimize) <= 1 and (len(smt.modinfo[topmod].maximize) + len(smt.modinfo[topmod].minimize)) <= 1)
```

Comment:
I think the last expression (len(..) + len(..) <= 1) implies the first two.

Reply:
It does in the real numbers, but there is the (perhaps irrelevantly small) chance of overflow which the full expression here avoids.